custom_components/keymaster/binary_sensor.py

```diff
@@ -78,7 +78,8 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
         _LOGGER.error("Z-Wave integration not found")
         raise PlatformNotReady
 
-    async_add_entities([entity], True)
+    if hass.states.get(f"binary_sensor.{slugify(ENTITY_NAME)}") is None:
```

Comment:
instead of having the guard here, let's move it up, and use the `ENTITY_ID` constant for the entity ID instead of regenerating it.

So at the top of the function:
```
if hass.states.get(ENTITY_ID):
  return True
```

Reply:
Adjusted in latest commit.